src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -720,8 +727,7 @@ public DynamicObject bytes(VirtualFrame frame, DynamicObject string, NotProvided
         }
 
         @Specialization
-        public DynamicObject bytes(DynamicObject string, DynamicObject block,
-                @Cached("create()") RopeNodes.BytesNode bytesNode) {
+        public DynamicObject bytes(DynamicObject string, DynamicObject block) {
```

Comment:
This (and below) is slightly different in terms of optimization as a single `RopeNodes.BytesNode` might be reused by both specializations (which could become more polymorphic than the original code if both specializations are taken and different specializations in `RopeNodes.BytesNode` are taken).
That's probably fine and worth the memory trade-off.
@nirvdrum Could you take a look since this is mostly your code?

Reply:
This case should be fine. I'll look at the others.

Reply:
They look fine to me. Most of it is moving profiles around for situations we never expect to happen anyway.

Reply:
@pitr-ch sent me a guide to run a simple rails application; I will use that to identify relevant patterns to optimize. The impact of the changes here in StringNode might be negligible, true.